kubectl-fdb/cmd/evacuate.go

```diff
@@ -0,0 +1,172 @@
+/*
+ * evacuate.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"log"
+
+	"k8s.io/apimachinery/pkg/fields"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/FoundationDB/fdb-kubernetes-operator/controllers"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ctx "context"
+
+	fdbtypes "github.com/FoundationDB/fdb-kubernetes-operator/api/v1beta1"
+)
+
+func newEvacuateCmd(streams genericclioptions.IOStreams, rootCmd *cobra.Command) *cobra.Command {
```

Comment:
What are the advantages to having a separate command for "evacuate", compared to adding more options to the "remove" command?

Reply:
It seemed to me the cleaner way since we don't remove the node we only remove all process groups running on it (actually I think we should name it `cordon` to have the same naming as Kubernetes.